packages/zipkin-instrumentation-grpc-client/src/grpcClientInstrumentation.js

```diff
@@ -0,0 +1,116 @@
+const {Annotation, HttpHeaders} = require('zipkin');
+
+/**
+ * @private
+ * @param {string} name
+ * @throws Error
+ */
+function requiredArg(name) {
+  throw new Error(`GrpcClientInstrumentation: Missing required argument ${name}.`);
+}
+
+/**
+ * @namespace grpc
+ */
+
+/**
+ * @typedef {Object} grpc.Status
+ * @memberof grpc
+ * @property {function()} clone
+ * @property {function(key: string, value: string)} add
+ */
+
+/**
+ * @typedef {Object} grpc.Metadata
+ * @memberof grpc
+ * @property {function()} clone
+ * @property {function(key: string, value: string)} add
+ */
+
+/**
+ * @typedef {Object} GrpcClientContext
+ * @property {zipkin.Tracer} tracer
+ * @property {string} serviceName
+ * @property {string} remoteServiceName
+ */
+
+/**
+ * @class GrpcClientInstrumentation
+ */
+class GrpcClientInstrumentation {
+  /**
+   * @constructor
+   * @param {Object} grpc
+   * @param {GrpcClientContext} context
+   */
+  constructor(grpc, {
+    tracer = requiredArg('tracer'),
+    serviceName = tracer.localEndpoint.serviceName,
```

Comment:
it isn't great to set this specific to a component as it encourages people to have inconsistent names. better to inherit it for now, from the tracer, so that it doesn't end up mixed.

remoteServiceName for client code is legit